packages/js/components/src/date-time-picker-control/date-time-picker-control.tsx

```diff
@@ -21,12 +22,17 @@ import {
 	__experimentalInputControl as InputControl,
 } from '@wordpress/components';
 
+export type DateTimePickerControlOnChangeHandler = (
+	date: string,
+	isValid: boolean
```

Comment:
Why we are exposing validations from inside the control? I thought validations are handling by the form validation machanism, outside of each control. 
I mean a DateControl should never emit an onChange with an incorrect (invalid formatted) value unless the value is `null | undefined`.
Also we should keep a standardize signature (as possible and respecting global props like value, onChange, onBlur and other common control props) for controls in order to increase reusability. Remember components like this will be used in conjunction with forms so a lot of adapter functions like `getInputProps` will be needed anytime a custom component like this arrives with a different props signature.

Reply:
While it is true that validation is handled invalid the form validation, there are a few reasons why validation is exposed from this component:

1. We are already doing the validation when passing the date to the DateTimePicker popup, so it is "free" for us to return whether it is valid or not (consumers can just ignore this parameter in the callback if they want)
2. If this component were to be used in a non-form scenario, consumers might want to just make use of this basic validation
3. By emitting `onChange` even for invalid values, consumers, like `<Form>` can use this to highlight when an invalid value is entered in the component. If `onChange` were only emitted for valid values, the consumer would not known when the component was in an invalid state.